sphinx/util/inventory.py

```diff
@@ -86,19 +87,23 @@ def load(
         reader = InventoryFileReader(stream)
         line = reader.readline().rstrip()
         if line == '# Sphinx inventory version 1':
-            return cls.load_v1(reader, uri, joinfunc)
+            invdata, ambiguities = cls.load_v1(reader, uri, joinfunc)
         elif line == '# Sphinx inventory version 2':
-            return cls.load_v2(reader, uri, joinfunc)
+            invdata, ambiguities = cls.load_v2(reader, uri, joinfunc)
         else:
             raise ValueError('invalid inventory header: %s' % line)
+        for ambiguity in ambiguities:
+            logger.warning(__("inventory <%s> contains multiple definitions for %s"),
+                           uri, ambiguity, type='intersphinx',  subtype='external')
+        return invdata
 
     @classmethod
     def load_v1(
         cls: type[InventoryFile],
         stream: InventoryFileReader,
         uri: str,
         join: Callable[[str, str], str],
-    ) -> Inventory:
+    ) -> tuple[Inventory, set[str]]:
```

Comment:
This is a breaking change to public API which, without deprecation warnings, I think is not ideal.

either, 

just log the warnings within `load_v2` (this seems easiest)

or you should create two methods for each version, like: `load_v2_with_ambiguties` (new signature), `load_v2` (calls `load_v2_with_ambiguties`, but then just drops the ambiguties)